app/filtering.js

```diff
@@ -474,50 +447,81 @@ function registerPermissionHandler (session, partition) {
       tempSettings = siteSettings.getSiteSettingsForURL(appState.get('temporarySiteSettings'), origin)
     }
 
-    const permissionName = permission + 'Permission'
-    let isAllowed
-    if (settings) {
-      isAllowed = settings.get(permissionName)
-    }
-    // Private tabs inherit settings from normal tabs, but not vice versa.
-    if (isPrivate && tempSettings) {
-      isAllowed = tempSettings.get(permissionName)
-    }
-    if (typeof isAllowed === 'boolean') {
-      cb(isAllowed)
-      return
-    }
+    let response = []
+    for (let i = 0; i < permissionTypes.length; i++) {
+      const permission = permissionTypes[i]
+      if (!permissions[permission]) {
+        console.warn('WARNING: got unregistered permission request', permission)
+        response.push(false)
+      }
 
-    // Display 'Brave Browser' if the origin is null; ex: when a mailto: link
-    // is opened in a new tab via right-click
-    const message = locale.translation('permissionMessage').replace(/{{\s*host\s*}}/, origin || 'Brave Browser').replace(/{{\s*permission\s*}}/, permissions[permission].action)
+      // The Torrent Viewer extension is always allowed to show fullscreen media
+      if (permission === 'fullscreen' &&
+        origin.startsWith('chrome-extension://' + config.torrentExtensionId)) {
+        response.push(true)
+      }
 
-    // If this is a duplicate, clear the previous callback and use the new one
-    if (permissionCallbacks[message]) {
-      permissionCallbacks[message](0, false)
-    }
+      // Always allow fullscreen if setting is ON
+      const alwaysAllowFullscreen = module.exports.alwaysAllowFullscreen() === fullscreenOption.ALWAYS_ALLOW
+      if (permission === 'fullscreen' && alwaysAllowFullscreen) {
+        response.push(true)
```

Comment:
if `permission === 'fullscreen'`, and both `origin.startsWith('chrome-extension://' + config.torrentExtensionId)` and `alwaysAllowFullscreen` are true, wouldn't you be calling `response.push(true)` twice in this loop (since the conditional is `if`, not `else if`)? `response[i] = ...` instead of `response.push(...)` seems clearer